routing/index_graph.cpp

```diff
@@ -172,12 +172,18 @@ RouteWeight IndexGraph::GetPenalties(Segment const & u, Segment const & v)
 {
   bool const fromPassThroughAllowed = m_geometry.GetRoad(u.GetFeatureId()).IsPassThroughAllowed();
   bool const toPassThroughAllowed = m_geometry.GetRoad(v.GetFeatureId()).IsPassThroughAllowed();
-
-  uint32_t const passThroughPenalty = fromPassThroughAllowed == toPassThroughAllowed ? 0 : 1;
-
-  if (IsUTurn(u, v))
-    return RouteWeight(m_estimator->GetUTurnPenalty(), passThroughPenalty, 0.0 /* transitTime */);
-
-  return RouteWeight(0.0, passThroughPenalty, 0.0 /* transitTime */);
+  // Route crosses border of pass-through/non-pass-through area if |u| and |v| have different
+  // pass through restrictions.
+  int8_t const passThroughPenalty = fromPassThroughAllowed == toPassThroughAllowed ? 0 : 1;
+
+  // We do not distinguish between RoadAccess::Type::Private and RoadAccess::Type::Destination for now.
+  bool const fromAccessAllowed = m_roadAccess.GetSegmentType(u) == RoadAccess::Type::Yes;
+  bool const toAccessAllowed = m_roadAccess.GetSegmentType(v) == RoadAccess::Type::Yes;
+  // Route crosses border of access=yes/access={private, destination} area if |u| and |v| have different
+  // access restrictions.
+  int8_t const accessPenalty = fromAccessAllowed == toAccessAllowed ? 0 : 1;
+
+  return RouteWeight(IsUTurn(u, v) ? m_estimator->GetUTurnPenalty() : 0.0 /* weight */,
```

Comment:
Can we extract this trenary to an individual method?

Reply:
```C++
auto const uTurnPenalty = IsUTurn(u, v) ? m_estimator->GetUTurnPenalty() : 0.0;

return RouteWeight(uTurnPenalty /* weight */, passThroughPenalty, accessPenalty, 0.0 /* transitTime */);
```

is it ok?